File: examples/php/read_tables.php
<?php
/**
 * Read an existing PDF and report the number of pages.
 *
 * Reads the file produced by generate_tables.php and prints the page count.
 * Run generate_tables.php first to create the input file.
 *
 * Mirrors: examples/rust/read_tables.rs
 *
 * Run with:
 *   php -d extension=target/release/libpdf_php.so examples/php/read_tables.php
 */

$path = __DIR__ . '/../output/php-tables.pdf';

if (!file_exists($path)) {
    fwrite(STDERR, "Error: file not found: $path\n");
    fwrite(STDERR, "Hint: run generate_tables.php first.\n");
    exit(1);
}

$reader = PdfReader::open($path);

echo "File:    $path\n";
echo "Version: PDF " . $reader->pdfVersion() . "\n";
echo "Pages:   " . $reader->pageCount() . "\n";

File: examples/rust/read_tables.rs
/// Example: Read an existing PDF and report the number of pages.
///
/// Reads the file produced by `generate_tables` and prints the page count.
/// Run `generate_tables` first to create the input file.
///
/// Run with:
///   cargo run --example read_tables -p pdf-examples
use pdf_core::PdfReader;

fn main() {
    let path = "examples/output/rust-tables.pdf";

    let reader = match PdfReader::open(path) {
        Ok(r) => r,
        Err(e) => {
            eprintln!("Error: {e}");
            eprintln!("Hint: run `cargo run --example generate_tables -p pdf-examples` first.");
            std::process::exit(1);
        }
    };

    println!("File:    {path}");
    println!("Version: PDF {}", reader.pdf_version());
    println!("Pages:   {}", reader.page_count());
}
